src/lib/encoder.ts

```diff
@@ -143,6 +143,109 @@ const decodeCompactBytesArray = (compactBytesArray: string): string[] => {
   return encodedValues;
 };
 
+const encodeBytesNCompactBytesArray = (
+  values: string[],
+  numberOfBytes: number,
+): string => {
+  values.forEach((value, index) => {
+    if (value.length > numberOfBytes * 2 + 2)
+      throw new Error(
+        `Hex bytes${numberOfBytes} value at index ${index} does not fit in ${numberOfBytes} bytes`,
+      );
+  });
+
+  return encodeCompactBytesArray(values);
+};
+
+const decodeBytesNCompactBytesArray = (
+  compactBytesArray: string,
+  numberOfBytes: number,
+): string[] => {
+  const bytesValues = decodeCompactBytesArray(compactBytesArray);
+  bytesValues.forEach((bytesValue, index) => {
+    if (bytesValue.length > numberOfBytes * 2 + 2)
+      throw new Error(
+        `Hex bytes${numberOfBytes} value at index ${index} does not fit in ${numberOfBytes} bytes`,
+      );
+  });
+
+  return bytesValues;
+};
+
+const returnTypesOfBytesNCompactBytesArray = () => {
+  const types = {};
+  for (let i = 1; i < 33; i++) {
+    types[`bytes${i}[CompactBytesArray]`] = {
+      encode: (value: string[]) => encodeBytesNCompactBytesArray(value, i),
+      decode: (value: string) => decodeBytesNCompactBytesArray(value, i),
+    };
+  }
+  return types;
+};
+
+const encodeUintNCompactBytesArray = (
+  values: number[],
+  numberOfBytes: number,
+): string => {
+  const hexValues: string[] = values.map((value, index) => {
+    const hexNumber = stripHexPrefix(numberToHex(value)).padStart(
+      numberOfBytes * 2,
+      '0',
+    );
+    if (hexNumber.length > numberOfBytes * 2)
+      throw new Error(
+        `Hex uint${
+          numberOfBytes * 8
+        } value at index ${index} does not fit in ${numberOfBytes} bytes`,
+      );
+    return hexNumber;
+  });
+
+  return encodeCompactBytesArray(hexValues);
+};
+
+const decodeUintNCompactBytesArray = (
+  compactBytesArray: string,
+  numberOfBytes: number,
+): number[] => {
+  const hexValues = decodeCompactBytesArray(compactBytesArray);
+
+  return hexValues.map((hexValue, index) => {
+    const hexValueStripped = stripHexPrefix(hexValue);
+    if (hexValueStripped.length > numberOfBytes * 2)
+      throw new Error(
+        `Hex uint${
+          numberOfBytes * 8
+        } value at index ${index} does not fit in ${numberOfBytes} bytes`,
+      );
+    return hexToNumber(hexValue);
+  });
+};
+
+const returnTypesOfUintNCompactBytesArray = () => {
```

Comment:
What about adding JSDocs to the functions, it's an open source project so I guess it's important to add JSDocs and comments for people to properly understand what's going on

Reply:
Yeah, good idea